### PR TITLE
Staking

### PR DESCRIPTION
**Pull request by pisarevkir**
Add Tooltip screen
Add Staking tooltip, Staking screen
Remove System screens

### CODE REVIEW

File: src/screens/Staking/components/StakingCalculator/components/Calculator.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+          {loading ? (
+            <ActivityIndicator />
+          ) : (
+            <Text
+              style={styles.resultValueText}
+              numberOfLines={1}
+              adjustsFontSizeToFit>
+              {result != null && (
+                <>
+                  {`${formatNumber(result)} ${t('mining_calculator.currency')}`}
```
Comment (victor-oancea-leftclick-io) [inline src/screens/Staking/components/StakingCalculator/components/Calculator.tsx:67]: any ice amount, anywhere in the app is string, always

File: src/screens/Staking/components/StakingCalculator/index.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  const [loading, setLoading] = useState(false);
+
+  const calculateResult = useCallback(
+    ({years, allocation}: {years: number; allocation: number}) => {
+      setLoading(true);
+      if (timer) {
+        clearTimeout(timer);
+      }
+      timer = setTimeout(() => {
+        setResult(years * allocation * 7120);
```
Comment (victor-oancea-leftclick-io) [inline src/screens/Staking/components/StakingCalculator/index.tsx:18]: you dont calculate anything, you just display it

File: src/screens/System/AppStateListener/index.tsx [hunk 3]
```diff
@@ -1,28 +0,0 @@
-// SPDX-License-Identifier: BUSL-1.1
-
```
Comment (victor-oancea-leftclick-io) [inline src/screens/System/AppStateListener/index.tsx:2]: why?

**Review by victor-oancea-leftclick-io — COMMENTED** [review]
why did you remove those system listeners?

File: src/screens/Staking/components/StakingCalculator/components/Calculator.tsx (hunk 1 above)
Comment (pisarevkir) [reply src/screens/Staking/components/StakingCalculator/components/Calculator.tsx:67]: i know, that was implemented just for illustration purposes, when we have BE integration, i'll remove those if we won't use it anywhere

File: src/screens/Staking/components/StakingCalculator/index.tsx (hunk 2 above)
Comment (pisarevkir) [reply src/screens/Staking/components/StakingCalculator/index.tsx:18]: yeap, that is just to show how loaders look, how the calendar feels with async calculation

File: src/screens/System/AppStateListener/index.tsx (hunk 3 above)
Comment (pisarevkir) [reply src/screens/System/AppStateListener/index.tsx:2]: this screen did do nothing - that was just a stub (we have `useAppStateListener` hook in `AppCommon` module that dispatches an event when the app changes the state so any other module can listen to it and do some stuff)

**Comment by pisarevkir** [comment]
> why did you remove those system listeners?

those were just stubs that do nothing

**Review by victor-oancea-leftclick-io — APPROVED** [review]

File: src/screens/Staking/components/StakingCalculator/components/Calculator.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+          />
+        </View>
+        <Slider
+          style={styles.slider}
+          progress={useSharedValue(ALLOCATION_DEFAULT)}
+          minimumValue={useSharedValue(ALLOCATION_MIN)}
+          maximumValue={useSharedValue(ALLOCATION_MAX)}
+          step={ALLOCATION_MAX}
+          onValueChange={value => {
+            allocationValueRef.current = Math.round(value); // https://0.30000000000000004.com/
```
Comment (OlehPanforov) [inline src/screens/Staking/components/StakingCalculator/components/Calculator.tsx:116]: What does this comment mean? 
` // https://0.30000000000000004.com/`

**Review by OlehPanforov — APPROVED** [review]

**Review by subbotnik16 — APPROVED** [review]

**Pull request merged**